### DISCUSSION TITLE
How can I tell if the onedrive docker container is syncing?

### DISCUSSION THREAD

**Discussion by everwisher** [abraunegg/onedrive#1993] [Q&A] [answered]
I set up an `abraunegg/ondrive` docker container and configured it to sync with one of my SharePoint site document folder. 

For testing, I used `docker exec -it onedrive /bin/bash` to enter the container and fired syncing manually using `onedrive --resync --monitor`. Since my files are very large and it would actually take a couple of days, if not weeks, to upload them all, and I can' leave a tty thread open to watch them uploading during that long period of time. So, I detached from the container but then I found no way to know if onedrive is still syncing in the background.

Is there a way to visualize the syncing progress, please?

**Comment by abraunegg** [comment, marked as answer]
@everwisher 
> For testing, I used docker exec -it onedrive /bin/bash to enter the container and fired syncing manually using onedrive --resync --monitor.

Dont do this and do not use 'resync' unless the application asks you to. To control your Docker instance either:
* Use the Docker environment variables: https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#environment-variables
* Edit the configuration as per https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#6-edit-the-config

> Is there a way to visualize the syncing progress, please?

There are three ways:
* Enable verbose logging
* Enable logging to a log file which would require configuration modification
* Use `docker logs onedrive` as per https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#4-docker-container-status-stop-and-restart - which will also show more data if verbose logging is enabled.

In all cases, make sure you are running an up-to-date Docker container.

**Reply by everwisher** [reply]
Thanks for replying! Before you replied I made another container on Synology DSM and through the visualization UI offered by Synology, I found the container wasn't syncing anything until I typed in `sudo docker exec -dit od1 onedrive --monitor` when it began to consume some cpu and ram. How can I make the container start to sync automatically at startup, please?

**Reply by abraunegg** [reply]
> Thanks for replying! Before you replied I made another container on Synology DSM and through the visualization UI offered by Synology, I found the container wasn't syncing anything until I typed in `sudo docker exec -dit od1 onedrive --monitor` when it began to consume some cpu and ram. How can I make the container start to sync automatically at startup, please?

The Docker containers *will* run the application correctly and automatically when they are started in the correct manner as per https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md

It sounds like your Synology is *not* starting the Docker container correctly.

Additionally, you are running the Docker container via sudo ..... this means you are running this as 'root' and you will run into other issues - do not do this ......

Please read the Docker documentation very carefully, and adapt this for your Synology environment. If you have difficulties - you need to seek support from Synology as to how to configure their platform to run the Docker container with the requirements that are detailed in the Docker documentation.

**Reply by everwisher** [reply]
I think Synology doesn't allow the command line way to kick start a container without `sudo`, but I used the environment variables to specify a normal user and group for running it. Anyway I will try it the UI way to build the container. Thanks for pointing this out!

**Reply by abraunegg** [reply]
Docker containers should not be run as 'root' .... the whole point of containers is to process isolate them from the host OS and each other to ensure there is a form of security boundary. 100% pointless if you are running everything as 'root' .... sounds to me like that is a bug / issue with Synology that you need to have fixed by Synology ...

**Reply by everwisher** [reply]
That's my bad about Docker. I take them only as containers for convenience of deployment and running but neglect the sense of isolation. Thanks for mentioning that to me! I think Synology just wants to force users to use their UI instead of CLI.

**Reply by abraunegg** [reply]
> Thanks for mentioning that to me! I think Synology just wants to force users to use their UI instead of CLI.

If you are going to use the CLI on your Synology - do not try and run Docker containers via CLI unless you get this working:

https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#0-install-docker-using-your-distribution-platforms-instructions
